cli/src/composer.rs

```diff
@@ -0,0 +1,211 @@
+use std::collections::HashMap;
+
+use failure::*;
+use rusqlite::Connection;
+use serde_json::Value;
+use trustnote::parent_composer::LastStableBallAndParentUnits;
+use trustnote::spec::{Message, Payload, Payment};
+use trustnote::*;
+
+struct Param {
+    signing_addresses: Vec<String>,
+    paying_addresses: Vec<String>,
+    outputs: Vec<spec::Output>,
+    messages: Vec<spec::Message>,
+    signer: String,
+    light_props: Option<LastStableBallAndParentUnits>,
```

Comment:
this should not be option but must, basically we should first prepare the praram and then compose the joint